### PR TITLE
cleanup mem & use true async to prevent freezing when execing cmds

### PR DESCRIPTION
**Pull request by nnyyxxxx**

### CODE REVIEW

**Comment by nnyyxxxx** [comment]
fixes swww not working when hyprpaper is running as well, we'll just killall previous backend if user switches

File: src/main.rs [hunk 1]
```diff
@@ -21,6 +22,9 @@ pub enum WallpaperBackend {
 }
 
 fn main() {
```
Comment (adamperkowski) [inline src/main.rs:24]: this wont work without

```suggestion
#[tokio::main]
 async fn main() {
```

**Comment by nnyyxxxx** [comment]
@adamperkowski thats causing panics

**Comment by adamperkowski** [comment]
@nnyyxxxx whats causing panics

**Comment by nnyyxxxx** [comment]
> @nnyyxxxx whats causing panics

https://github.com/nnyyxxxx/hyprwall/pull/10#discussion_r1805880622

**Comment by nnyyxxxx** [comment]
ready to merge ? @adamperkowski

File: src/main.rs (hunk 1 above)
Comment (nnyyxxxx) [reply src/main.rs:24]: not necessary with the runtime below this, also with how gtk works doing it your way will cause panics

**Comment by adamperkowski** [comment]
i think so. sorry for the last suggestion, misread the code im on my phone

**Comment by nnyyxxxx** [comment]
waot, i noticed a bug

**Comment by nnyyxxxx** [comment]
nvm, false alarm

**Pull request merged**